src/Homepage/Button/SearchButton.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+import './_search-button.scss';
+
+import { Link } from 'react-router-dom';
+
+
+
+export default class SearchButton extends Component {
+
+    shouldComponentUpdate(nextProps, nextState, nextContext) {
+        return this.props.query !== nextProps.query || this.props.selectedOption !== nextProps.selectedOption;
```

Comment:
You can destruct `this.props.query` and `this.props.selectedOption` above in the function and then use it.